python/ribasim/ribasim/schemas.py

```diff
@@ -16,9 +16,12 @@ class Config:
         add_missing_columns = True
         coerce = True
 
-    @classmethod
-    def _index_name(self) -> str:
-        return "fid"
+    @pa.dataframe_parser
+    def _name_index(cls, df):
+        # Node and Edge have different index names, avoid running both parsers
+        if cls.__name__ not in ("NodeSchema", "EdgeSchema"):
+            df.index.name = "fid"
+        return df
```

Comment:
I was confused for a while since I expected the methods in the subclasses in edge.py and node.py to override this. But when debugging a test failure I saw that first the node parser was applied, naming it "node_id", and then this one, making it end up with the wrong name. Hence the check for those class names.

Reply:
How about `df.index.name = cls._index_name()`? A mix of both old and new.

Reply:
Yeah we could go with that as well.

Reply:
Done in 007ee67bc71cf3491dea67d94078be0bf2d9a15d